src/main/doc/authenticationMechanism.asciidoc

```diff
@@ -330,60 +330,54 @@ public @interface LoginToContinue {
 }
 ----
 
-The container MUST provide an interceptor implementation at priority _PLATFORM_BEFORE + 220_ that backs the _LoginToContinue_ annotation and intercepts calls to the configured _HttpAuthenticationMechanism_. The interceptor MUST behave as follows when intercepting calls to the _HttpAuthenticationMechanism_:
+The container MUST provide an interceptor implementation, at priority _PLATFORM_BEFORE_ + 220, that backs the _LoginToContinue_ annotation and intercepts calls to the configured _HttpAuthenticationMechanism_. The interceptor MUST behave as follows when intercepting calls to the _HttpAuthenticationMechanism_:
 
 Intercepting _validateRequest()_::
-* Determine if there's any state lingering behind from a previously aborted flow involving "login to continue".
-* If there is:
-** Determine if the caller aborted the earlier flow, and now does a new request to a protected resource or initiated a new authentication dialog explicitly (i.e. by calling _SecurityContext#authenticate_ with _AuthenticationParameters#newAuthentication_ set to _true_).
-** If so:
-*** Clean (remove) all state related to "login to continue" and create new state that initially records whether the caller did a request to a protected resource or initiated a new authentication dialog for the current request.
-* Use the state to determine whether the caller did a request to a protected resource or initiated a new authentication dialog (note: not necessarily in the current request).
-* If the caller initiated a new authentication dialog, then:
-** Continue with flow "processCallerInitiatedAuthentication".
-* Otherwise
-** Continue with flow "processContainerInitiatedAuthentication".
-
-Flow "processCallerInitiatedAuthentication"::
-* Call the next Interceptor. For the sake of discussion call its outcome _outcome_.
-* If _outcome_ equals _SUCCESS_ and _HttpMessageContext#getCallerPrincipal()_ is not null, remove all state.
-* Return _outcome_.
-
-Flow "processContainerInitiatedAuthentication"::
+* Determine if there is any stale state in the request context, due to a previously aborted flow involving "login to continue". If so, clear the stale state.
+* Determine if this request is a new caller-initiated authentication, by calling _isNewAuthentication()_ on the _AuthenticationParameters_ object available from _HttpMessageContext_.
+** If _isNewAuthentication()_ returns true, update the request state to record that it is a caller-initiated authentication, and continue with flow _processCallerInitiatedAuthentication_.
+** If _isNewAuthentication()_ returns false, continue with flow _processContainerInitiatedAuthentication_.
+
```

Comment:
OK, I see what you mean. The decision should be based on the value of `isCallerInitiatedAuthentication()`. I did look at the code, but wrote it that way because your original writeup based the decision on "If the caller initiated a new authentication dialog".

The other part is pulling the check of `isNewAuthentication()` and the call to `saveCallerInitiatedAuthentication()` out of the clean logic. It's efficient to do that in `tryClean()`, because you only have to check `isNewAuthentication()` once, but it seemed more logical to me to make it a separate step, and that logic still seems to work.

I'll change the language about using `newAuthentication` to choose the flow.